lib/logstash/pluginmanager/install.rb

```
@@ -1,82 +1,128 @@
-require 'logstash/pluginmanager'
 require 'logstash/pluginmanager/util'
-require 'rubygems/dependency_installer'
-require 'rubygems/uninstaller'
 require 'jar-dependencies'
 require 'jar_install_post_install_hook'
 require 'file-dependencies/gem'
 
+require "logstash/gemfile"
+require "logstash/bundler"
+
 class LogStash::PluginManager::Install < Clamp::Command
+  parameter "[PLUGIN] ...", "plugin name(s) or file"
+  option "--version", "VERSION", "version of the plugin to install"
+  option "--force", :flag, "force install without verifying plugin validity"
+  option "--development", :flag, "install all development dependencies of currently installed plugins"
 
-  parameter "PLUGIN", "plugin name or file"
+  # the install logic below support installing multiple plugins with each a version specification
+  # but the argument parsing does not support it for now so currently if specifying --version only
+  # one plugin name can be also specified.
+  #
+  # TODO: find right syntax to allow specifying list of plugins with optional version specification for each
 
-  option "--version", "VERSION", "version of the plugin to install", :default => ">= 0"
+  def execute
+    if development?
+      unless plugin_list.empty?
+        $stderr.puts("Cannot specify plugin(s) with --development, it will add the development dependencies of the currently installed plugins")
+        exit(1)
+      end
+    else
+      if plugin_list.empty? && !force?
+        $stderr.puts("No plugin specified")
+        exit(1)
+      end
+      if version && plugin_list.size > 1
+        # temporary until we fullfil TODO ^^
+        $stderr.puts("Only 1 plugin name can be specified with --version")
+        exit(1)
+      end
+    end
 
-  option "--proxy", "PROXY", "Use HTTP proxy for remote operations"
+    unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+      $stderr.puts("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting")
+      exit(1)
```

my thought is that we should raise an exception that will trigger an `exit(1)` by clamp. this is the right usage for exceptions and having "artificial" integer return codes for the `execute` method feels silly.